datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -1024,11 +1017,7 @@ impl GroupedHashAggregateStream {
     /// Note: currently spilling is not supported for Partial aggregation
     fn update_skip_aggregation_probe(&mut self, input_rows: usize) {
         if let Some(probe) = self.skip_aggregation_probe.as_mut() {
-            if !self.spill_state.spills.is_empty() {
```

Comment:
Now partial aggregation only do early emit indeed, but I guess this check is trying to be defensive, if someone decides to do spilling in the partial stage also, the early emit logic won't be break 🤔 Maybe we can leave an assertion here?

Reply:
In https://github.com/apache/datafusion/pull/7400, the author tried spilling in partial stage, but he was asked to remove that. See https://github.com/apache/datafusion/pull/7400#discussion_r1306072986.

Reply:
> I guess this check is trying to be defensive

Yes, that was the idea -- these features are mutually exclusive, with current implementation of spilling, but if they both are triggered (which never happens for now -- probably the [note](https://github.com/apache/datafusion/blob/cad41464fb2be574bfe9e72b4279c56ec0ebd132/datafusion/physical-plan/src/aggregates/row_hash.rs#L1024) should be more informative) it shouldn't break query execution.

I think the change is reasonable since current code is redundant indeed, but not sure about assertion -- I suppose it's better to return not_implemented / internal error instead of panicking.

And it's probably worth to retain the note on why skip partial is incompatible with spilling.

Reply:
Should we add another assert to ensure and emphasize that the `update_state` is only called in `Partial` mode?

```
               ...
               assert!(self.spill_state.spills.is_empty() && self.mode == AggregateMode::Partial);
               probe.update_state(input_rows, self.group_values.len());
              ....

```

Reply:
I agree -- adding an assertion would be great. Can you perhaps make a PR to do so?

Reply:
sure

Reply:
I have submitted a pr about this https://github.com/apache/datafusion/pull/12640